### PR TITLE
Option for choosing a groupsprovider

### PR DESCRIPTION
**Pull request by preemeijer**
Hello,

An option made available so that a user can choose for POSIX groups.

Greetings, Paul

### CODE REVIEW

File: src/fkooman/VootProvider/LdapVootStorage.php [hunk 1]
```diff
@@ -127,6 +129,19 @@ public function getGroupMembers($resourceOwnerId, $groupId, $startIndex = 0, $co
         if (false === $query) {
             throw new VootStorageException("ldap_error", "directory query for group failed");
         }
+	
+        $all = ldap_get_entries($this->ldapConnection, $query); 
+        
+        switch ($groupsProvider):
+	        case "posixgroup":
+		        $all = ldap_get_entries($this->ldapConnection, $query);
```
Comment (thijskh) [inline src/fkooman/VootProvider/LdapVootStorage.php]: Are we now performing the same LDAP query twice when using `posixgroup`?

File: src/fkooman/VootProvider/LdapVootStorage.php [hunk 2]
```diff
@@ -167,8 +191,19 @@ public function isMemberOf($resourceOwnerId, $startIndex = null, $count = null)
...
         /* get the groups the user is a member of */
-        $filter = '(' . $this->config->s('LdapVootStorage')->l('memberAttribute') . '=' . $userDn . ')';
+        switch ($groupsProvider):
+            case "posixgroup":
+                $filter = '(' . $this->config->s('LdapVootStorage')->l('memberAttribute') . '=' . $resourceOwnerId . ')';
+            break;
+            default:
+                $filter = '(' . $this->config->s('LdapVootStorage')->l('memberAttribute') . '=' . $userDn . ')';
+            break;
+        endswitch;
```
Comment (thijskh) [inline src/fkooman/VootProvider/LdapVootStorage.php]: Can you use `{` and `}` to define blocks to be consistent with the other code?

I'm wondering if `switch` really adds value here or that just using if/else would suffice.

**Comment by thijskh** [comment]
Thanks! In principle this looks good. I've added some comments that might improve it some more.

File: src/fkooman/VootProvider/LdapVootStorage.php (hunk 2 above)
Comment (preemeijer) [inline src/fkooman/VootProvider/LdapVootStorage.php]: The idea behind using  switch is that you have the possibility to add other group providers more easily. I think this is also somewhat personal to choose between the two.

Comment (thijskh) [inline src/fkooman/VootProvider/LdapVootStorage.php]: Alright.

**Comment by thijskh** [comment]
Thanks for the fixes, I've merged it now.

**Pull request merged**